state-chain/pallets/cf-swapping/src/tests.rs

```diff
@@ -1827,6 +1818,49 @@ fn max_swap_amount_can_be_removed() {
 	});
 }
 
+#[test]
+fn swap_input_excludes_network_fee() {
+	const AMOUNT: AssetAmount = 1_000;
+	const FROM_ASSET: Asset = Asset::Usdc;
+	const TO_ASSET: Asset = Asset::Flip;
+	let destination_address: ForeignChainAddress = ForeignChainAddress::Eth(Default::default());
+	const NETWORK_FEE: Percent = Percent::from_percent(1);
+
+	NetworkFee::set(NETWORK_FEE);
+
+	new_test_ext()
+		.execute_with(|| {
+			Swapping::schedule_swap_from_channel(
+				ForeignChainAddress::Eth(Default::default()),
+				0,
+				FROM_ASSET,
+				TO_ASSET,
+				AMOUNT,
+				destination_address.clone(),
+				bounded_vec![],
+				0,
+			);
+		})
+		.then_process_blocks_until(|_| System::block_number() == 3)
+		.then_execute_with(|_| {
+			let expected_swap_input = AMOUNT - NETWORK_FEE * AMOUNT;
```

Comment:
I think this might be where I got confused.

In a two-legged swap, the swap input (of the whole swap) is pre-network-fee. 
But in a one-legged swap we take the network fee and then emit the swap input... 

IMO this is another symptom of conflating 'user swaps' and 'amm swaps'/'swap legs'.

I think we should structure it more like this going forward: 

swap_input
first_leg: Optional [
	swap_leg_input
		-> LP Fee
	swap_leg_output
]
	-> network_fee
second_leg: Optional [
	swap_leg_input
		-> LP Fee
	swap_leg_output
]
swap_output